fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -342,57 +404,74 @@ fi
 
 log_Msg "mkdir -p ${fMRIFolder}/MotionCorrection"
 mkdir -p "$fMRIFolder"/MotionCorrection
+
 ${RUN} "$PipelineScripts"/MotionCorrection.sh \
        "$fMRIFolder"/MotionCorrection \
        "$fMRIFolder"/"$NameOffMRI"_gdc \
-       "$fMRIFolder"/"$ScoutName"_gdc \
+       "$fMRIReferenceImage" \
        "$fMRIFolder"/"$NameOffMRI"_mc \
        "$fMRIFolder"/"$MovementRegressor" \
        "$fMRIFolder"/"$MotionMatrixFolder" \
        "$MotionMatrixPrefix" \
-       "$MotionCorrectionType"
+       "$MotionCorrectionType" \
+       "$fMRIReferenceReg"
 
 # EPI Distortion Correction and EPI to T1w Registration
-log_Msg "EPI Distortion Correction and EPI to T1w Registration"
-
 DCFolderName=DistortionCorrectionAndEPIToT1wReg_FLIRTBBRAndFreeSurferBBRbased
 DCFolder=${fMRIFolder}/${DCFolderName}
 
-if [ -e ${DCFolder} ] ; then
-    ${RUN} rm -r ${DCFolder}
+if [ $fMRIReference = "NONE" ] ; then
+  log_Msg "EPI Distortion Correction and EPI to T1w Registration"
+
+  if [ -e ${DCFolder} ] ; then
+      ${RUN} rm -r ${DCFolder}
+  fi
+  log_Msg "mkdir -p ${DCFolder}"
+  mkdir -p ${DCFolder}
+
+  ${RUN} ${PipelineScripts}/DistortionCorrectionAndEPIToT1wReg_FLIRTBBRAndFreeSurferBBRbased.sh \
+         --workingdir=${DCFolder} \
+         --scoutin=${fMRIFolder}/${ScoutName}_gdc \
+         --t1=${T1wFolder}/${T1wImage} \
+         --t1restore=${T1wFolder}/${T1wRestoreImage} \
+         --t1brain=${T1wFolder}/${T1wRestoreImageBrain} \
+         --fmapmag=${MagnitudeInputName} \
+         --fmapphase=${PhaseInputName} \
+         --fmapgeneralelectric=${GEB0InputName} \
+         --echodiff=${deltaTE} \
+         --SEPhaseNeg=${SpinEchoPhaseEncodeNegative} \
+         --SEPhasePos=${SpinEchoPhaseEncodePositive} \
+         --echospacing=${EchoSpacing} \
+         --unwarpdir=${UnwarpDir} \
+         --owarp=${T1wFolder}/xfms/${fMRI2strOutputTransform} \
+         --biasfield=${T1wFolder}/${BiasField} \
+         --oregim=${fMRIFolder}/${RegOutput} \
+         --freesurferfolder=${T1wFolder} \
+         --freesurfersubjectid=${Subject} \
+         --gdcoeffs=${GradientDistortionCoeffs} \
+         --qaimage=${fMRIFolder}/${QAImage} \
+         --method=${DistortionCorrection} \
+         --topupconfig=${TopupConfig} \
+         --ojacobian=${fMRIFolder}/${JacobianOut} \
+         --dof=${dof} \
+         --fmriname=${NameOffMRI} \
+         --subjectfolder=${SubjectFolder} \
+         --biascorrection=${BiasCorrection} \
+         --usejacobian=${UseJacobian}
+else
+    log_Msg "linking EPI distortion correction and T1 registration from ${fMRIReference}"
+    if [ -e ${fMRIFolder}/${DCFolderName} ] ; then
+        log_Warn "     ... removing stale link (or preexisiting files)"
+        rm -r ${fMRIFolder}/${DCFolderName}
+    fi
+    ln -s ${fMRIReference}/${DCFolderName} ${fMRIFolder}/${DCFolderName}
+ 
+    WD=${fMRIReference}/${DCFolderName}
+    ${FSLDIR}/bin/imcp ${fMRIReference}/${RegOutput} ${fMRIFolder}/${RegOutput}
```

Comment:
So, ${RegOutput} is named "Scout2T1w".  Is that going to lead to ambiguity regarding which particular "Scout" it is referring to?

Reply:
I thought that the file is used downstream and it has to keep that name, but it apparently is not, so I changed it to: 

```
${FSLDIR}/bin/imcp ${fMRIReference}/${RegOutput} ${fMRIFolder}/Reference${RegOutput}
```

So that it is clear that it is the reference scout.